spec/features/user_search_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+describe "Filter users by key", type: :feature do
+
+     before(:each) do
+        @user1 = User.new(firstname:"Max", lastname:"Mustermann", username:"Maxi", email: "example@example.com")
+        @user2 = User.new(firstname:"Grace", lastname:"Hopper", username:"Profi", email: "demo@demo.com")
+        sign_in @user1
+      end
+
+    it "should be possible to filter by firstname" do
+        visit users_search_path(firstname: "Ma")
+        expect(page).to have_text("Max Mustermann")
+        expect(page).not_to have_text("Grace")
+    end
+
+    it "should be possible to filter by lastname" do
+       visit users_search_path(lastname: "mann")
+       expect(page).to have_text("Max Mustermann")
+       expect(page).not_to have_text("Grace")
+    end
+
+    it "should be possible to filter by username" do
+        visit users_search_path(user: "Ma")
```

Comment:
Because the firstname also starts with `Ma` it cannot be assured that it's really the username that is used for filtering

Reply:
Right